pkg/nsx/services/subnetport/subnetport.go

```diff
@@ -569,17 +569,21 @@ func (service *SubnetPortService) AllocatePortFromSubnet(subnet *model.VpcSubnet
 		}
 	}
 
+	// For DHCP Deactivated mode Subnet, if staticIpAllocation enable:false, skip check IP count
+	// and always return true
```

Comment:
I feel we can move the part to the beginning of the function as no ip Subnet do not need to check the cache we do not need to store it in cache or lock the cache.